sktime/param_est/compose.py

```diff
@@ -301,3 +302,120 @@ def get_test_params(cls, parameter_set="default"):
             params = params + [{"transformers": [t1, t2], "param_est": p}]
 
         return params
+
+
+class FunctionParamFitter(BaseParamFitter):
+    r"""Constructs a parameter fitter from an arbitrary callable.
+
+    A FunctionParamFitter forwards its X argument to a user-defined
+    function (or callable object) and sets the result of this function
+    to the ``param`` attribute. This can be useful for stateless
+    estimators such as simple conditional parameter selectors.
+
+    Note: If a lambda function is used as the ``func``, then the
```

Comment:
yes, indeed, it is sth I wanted to discuss.

For now, the design makes sense, I just pointed out that you can define a custom `save` function and `load_from_` functions.

We have a couple other cases where `lambda`-s can be parameters, and we have not found a good way to make these serializable. In case you have such good ideas, we are all ears. FYI @achieveordie

Reply:
Without having thought too much about this, I am initially inclined to say that this does not sound like something that sktime should try to solve. 

I believe that there are mature tools out there that _could_ be used to serialise lambdas (for instance, https://github.com/uqfoundation/dill). After a quick Google search, I saw that even using the builtin `marcal` package, you can do a simple round trip with:
```pycon
>>> my_lambda = lambda: "Hello"
>>> my_lambda_roundtrip = types.FunctionType(marshal.loads(marshal.dumps(my_lambda.__code__)), globals())
>>> my_lambda_roundtrip()
'Hello'
```

However, I could not find a reliable (**and simple!**) way to **make** a lambda pickleable.

Even if there was it sounds strange for sktime to do this behind the scenes even when it might not be necessary to do so, adding unnecessary overhead to all constructors that take callables as arguments.

Resolving this comment for now as it is not relevant to this PR but interesting in a follow up on this 👍